staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -918,8 +866,7 @@ func (r *Request) request(ctx context.Context, fn func(*http.Request, *http.Resp
 				fn(req, resp)
 			}
 
-			var retry bool
-			retryAfter, retry = r.retry.NextRetry(req, resp, err, func(req *http.Request, err error) bool {
+			if retry := r.retry.IsNextRetry(ctx, r, req, resp, err, func(req *http.Request, err error) bool {
```

Comment:
nit: can you define the input function excplitly (same as in line 605 for Watch()) ?

Reply:
as soon as I define an explicit func, `TestDoRequestSuccess` keeps failing, it's very strange, I will pursue it in a follow up PR if that is okay.

Reply:
i had a typo, it's fixed now.